eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version3/ChannelCodecs3.scala

```diff
@@ -387,14 +414,15 @@ private[channel] object ChannelCodecs3 {
 
   // Order matters!
   val channelDataCodec: Codec[PersistentChannelData] = discriminated[PersistentChannelData].by(uint16)
+    .typecase(0x09, Codecs.DATA_NORMAL_Codec)
     .typecase(0x08, Codecs.DATA_SHUTDOWN_Codec)
-    .typecase(0x07, Codecs.DATA_NORMAL_Codec)
+    .typecase(0x07, Codecs.DATA_NORMAL_COMPAT_07_Codec)
     .typecase(0x06, Codecs.DATA_WAIT_FOR_REMOTE_PUBLISH_FUTURE_COMMITMENT_Codec)
     .typecase(0x05, Codecs.DATA_CLOSING_Codec)
     .typecase(0x04, Codecs.DATA_NEGOTIATING_Codec)
     .typecase(0x03, Codecs.DATA_SHUTDOWN_COMPAT_03_Codec)
     .typecase(0x02, Codecs.DATA_NORMAL_COMPAT_02_Codec)
-    .typecase(0x01, Codecs.DATA_WAIT_FOR_FUNDING_LOCKED_Codec)
+    .typecase(0x01, Codecs.DATA_WAIT_FOR_CHANNEL_READY_Codec)
```

Comment:
You forgot to add DATA_WAIT_FOR_CHANNEL_READY_COMPAT_01_Codec, we probably need a migration test on this!

Reply:
Dammit! Nice catch. I would probably have caught it when deploying on endurance, the ultimate migration test.

Reply:
Fixed in 897f5f1b9.